src/deploy-cromwell-on-azure/Deployer.cs

```diff
@@ -583,9 +694,302 @@ bool MatchesClientId(Task<IIdentity> identity)
+            {
+                await PatchContainersToMountFileV210Async(storageAccount, managedIdentity.Name);
+            }
+
+            if (installedVersion == null || installedVersion < new Version(2, 2))
+            {
+                await PatchContainersToMountFileV220Async(storageAccount);
+            }
+
+            if (installedVersion == null || installedVersion < new Version(2, 4))
+            {
+                await PatchContainersToMountFileV240Async(storageAccount);
+                await PatchAccountNamesFileV240Async(sshConnectionInfo, managedIdentity);
+            }
+
+            if (installedVersion == null || installedVersion < new Version(2, 5))
+            {
+                await MitigateChaosDbV250Async(cosmosDb);
+            }
+
+            if (installedVersion == null || installedVersion < new Version(3, 0))
+            {
+                await PatchCromwellConfigurationFileV300Async(storageAccount);
+                await AddNewSettingsV300Async(sshConnectionInfo);
+                await UpgradeBlobfuseV300Async(sshConnectionInfo);
+                await DisableDockerServiceV300Async(sshConnectionInfo);
+
+                ConsoleEx.WriteLine($"It's recommended to update the default CoA storage account to a General Purpose v2 account.", ConsoleColor.Yellow);
+                ConsoleEx.WriteLine($"To do that, navigate to the storage account in the Azure Portal,", ConsoleColor.Yellow);
+                ConsoleEx.WriteLine($"Configuration tab, and click 'Upgrade.'", ConsoleColor.Yellow);
+            }
+
+            if (installedVersion is null || installedVersion < new Version(3, 1))
+            {
+                await PatchCromwellConfigurationFileV310Async(storageAccount);
+            }
+        }
+
+        private Dictionary<string, string> GetSystemSettings()
+        {
+            var settings = new Dictionary<string, string>
+            {
+                ["BatchNodesSubnetId"] = configuration.BatchNodesSubnetId,
+                ["DockerInDockerImageName"] = configuration.DockerInDockerImageName,
+                ["BlobxferImageName"] = configuration.BlobxferImageName,
+                ["DisableBatchNodesPublicIpAddress"] = configuration.DisableBatchNodesPublicIpAddress.GetValueOrDefault().ToString(),
+                ["KeepSshPortOpen"] = configuration.KeepSshPortOpen.GetValueOrDefault().ToString()
+            };
+
+            return settings;
+        }
+
+        private Dictionary<string, string> GetPersonalizedSettings(IIdentity managedIdentity)
+        {
+            var files = new string[] { "env-00-coa-version.txt", "env-01-account-names.txt", "env-02-internal-images.txt", "env-03-external-images.txt", "env-04-settings.txt" };
+            var settings = new Dictionary<string, string>();
+
+            foreach (var file in files)
+            {
+                var fileContents = Utility.GetFileContent("scripts", file);
```

Comment:
Use Utility.DelimitedTextToDictionary() to read settings from each file, then join the resulting dictionaries using Union() 

foreach (var file in files)
            {
                settings = settings.Union(Utility.DelimitedTextToDictionary(Utility.GetFileContent("scripts", file))).ToDictionary(kv => kv.Key, kv => kv.Value);
                }
                
                Or add and overload to Utility.DelimitedTextToDictionary to take IEnumerable<string> as the first parameter and then pass array of file contents to it.

Reply:
Fixed